stores/current.ts

```diff
@@ -159,29 +159,51 @@ function round2(value: number): number {
 }
 
 function round4(value: number): number {
+  if (typeof value !== 'number') {
+    throw new TypeError(`Expected a number but received ${typeof value}`);
+  }
+
   return parseFloat(value.toFixed(4))
 }
 
-function roundValue<V extends Partial<LchValue>>(
-  value: V,
-  type: 'lch' | 'oklch'
-): V {
-  let rounded = { ...value }
-  if (typeof rounded.l !== 'undefined') {
-    rounded.l = round2(rounded.l)
-  }
-  if (typeof rounded.c !== 'undefined') {
-    rounded.c = type === 'oklch' ? round4(rounded.c) : round2(rounded.c)
-  }
-  if (typeof rounded.h !== 'undefined') {
-    rounded.h = round2(rounded.h)
-  }
-  if (typeof rounded.a !== 'undefined') {
-    rounded.a = round2(rounded.a)
+
+function incrementalRoundValue(value: LchValue, COLOR_FN: string): LchValue {
+  const maxIterations = 6;
+
+  for (let i = 0; i <= maxIterations; i++) {
+    let precision = i === 0 ? 0 : Math.min(i, maxIterations);
+    let roundedValue: LchValue = {
+      a: round2(value.a),
+      c: round2(value.c),
+      h: round2(value.h),
+      l: round2(value.l),
+    };
+
+    roundedValue.l = parseFloat(roundedValue.l.toFixed(precision));
+    roundedValue.c = parseFloat(roundedValue.c.toFixed(precision));
+    roundedValue.h = parseFloat(roundedValue.h.toFixed(precision));
+    roundedValue.a = parseFloat(roundedValue.a.toFixed(precision));
+
+    if (isMatch(roundedValue)) {
+      return roundedValue;
+    }
   }
-  return rounded
+
+  return value;
+}
+
+function isMatch(value: LchValue): boolean {
```

Comment:
1. You check only the fact that sRGB didn’t go to P3. But we need also to check that P3 color will not go to Rec2020. Check the origin algorithm https://github.com/evilmartians/oklch-picker/blob/main/stores/current.ts#L196
2. Do we need to replace the origin algorithm? Seems like right now we have 2 different checks https://github.com/evilmartians/oklch-picker/blob/main/stores/current.ts#L195-L200